pywbemtools/pywbemcli/_cmd_listener.py

```diff
@@ -0,0 +1,473 @@
+
+
+@cli.group('listener', cls=PywbemcliGroup, options_metavar=GENERAL_OPTS_TXT,
+           subcommand_metavar=SUBCMD_HELP_TXT)
+@add_options(help_option)
+def listener_group():
+    """
+    Command group for WBEM indication listeners.
+
+    This command group defines commands to manage WBEM indication listeners.
+    Each listener is a running process that executes the
+    `pywbemcli listener run` command acting as a listener.
+
+    These processes can be started manually by the user by invoking the
+    `pywbemcli listener run` command in the background, or by using the
+    `pywbemcli listener start` command which does exactly that.
+
+    There is no persisted data about the currently running listeners, instead
+    the currently running processes executing the `pywbemcli listener run`
+    command are the currently running listeners.
+
+    If such a listener process terminates, the corresponding listener no longer
+    exists, so there is no notion of a stopped listener nor does a listener
+    have an operational status. The `pywbemcli listener stop` command terminates
+    a listener gracefully.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'pywbemcli --help') can also be specified before the
+    'connection' keyword.
+    """
+    pass  # pylint: disable=unnecessary-pass
+
+
+@listener_group.command('run', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(LISTEN_OPTIONS)
+@add_options(help_option)
+@click.pass_obj
+def listener_run(context, **options):
+    """
+    Run as a named WBEM indication listener.
+
+    Run this command as a named WBEM indication listener until it gets
+    terminated, e.g. by a keyboard interrupt, break signal (e.g. kill), or the
+    `pywbemcli listener stop` command.
+
+    A listener with that name must not be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener run lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_run(context, options))
+
+
+@listener_group.command('start', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(LISTEN_OPTIONS)
+@add_options(help_option)
+@click.pass_obj
+def listener_start(context, **options):
+    """
+    Start a named WBEM indication listener in the background.
+
+    A listener with that name must not be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener start lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_start(context, options))
+
+
+@listener_group.command('stop', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(help_option)
+@click.pass_obj
+def listener_stop(context, name):
```

Comment:
This one could be extended to present a list of running listeners from which one is selected to stop

Reply:
Is that really needed? How many listeners do you expect?

Reply:
Lets ignore my comment at least until we better understand the listener